tests/syslog/test_logrotate.py

```diff
@@ -0,0 +1,211 @@
+def disable_logrotate_cron_job(rand_selected_dut):
+    with DisableLogrotateCronContext(rand_selected_dut):
+        yield
+
+
+@pytest.fixture(scope='module', autouse=True)
+def backup_syslog(rand_selected_dut):
+    """
+    Back up current syslog file
+    :param rand_selected_dut: The fixture returns a randomly selected DUT
+    """
+    duthost = rand_selected_dut
+    logger.info('Backup syslog file to syslog_bk')
+    duthost.shell('sudo cp -f /var/log/syslog /var/log/syslog_bk')
+
+    yield
+
+    logger.info('Recover syslog file to syslog')
+    duthost.shell('sudo mv /var/log/syslog_bk /var/log/syslog')
+
+    logger.info('Remove temp file /var/log/syslog.1')
+    duthost.shell('sudo rm -f /var/log/syslog.1')
+
+    logger.info('Restart rsyslog service')
+    duthost.shell('sudo service rsyslog restart')
+
+
+@pytest.fixture(scope='function')
+def simulate_small_var_log_partition(rand_selected_dut, localhost):
+    """
+    Simulate a small var log partition
+    :param rand_selected_dut: The fixture returns a randomly selected DUT
+    """
+    duthost = rand_selected_dut
+    logger.info('Create a small var log partition with size of {}'.format(SMALL_VAR_LOG_PARTITION_SIZE))
+    duthost.shell('sudo fallocate -l {} log-new-partition'.format(SMALL_VAR_LOG_PARTITION_SIZE))
+    duthost.shell('sudo losetup -P  /dev/loop2 log-new-partition')
+    duthost.shell('sudo mkfs.ext4 /dev/loop2')
+    duthost.shell('sudo mount /dev/loop2 /var/log')
+
+    config_reload(duthost, safe_reload=True)
+
+    logger.info('Start logrotate-config service')
+    duthost.shell('sudo service logrotate-config start')
+
+    yield
+
+    logger.info('Umount and unload the small var log partition')
+    duthost.shell('sudo umount -l /dev/loop2')
+    duthost.shell('sudo losetup -d /dev/loop2')
+
+    logger.info('Remove the small var log partition')
+    duthost.shell('sudo rm -f log-new-partition')
+
+    config_reload(duthost, safe_reload=True)
+
+    logger.info('Restart logrotate-config service')
+    duthost.shell('sudo service logrotate-config restart')
+
+
+def get_var_log_size(duthost):
+    """
+    Check the size of /var/log folder
+    :param duthost: DUT host object
+    :return: size value
+    """
+    size = duthost.shell("sudo df -k /var/log | sed -n 2p | awk '{ print $2 }'")['stdout']
+    return int(size)
+
+
+def get_syslog_file_count(duthost):
+    """
+    Check the rotated syslog file number
+    :param duthost: DUT host object
+    :return: file number value
+    """
+    logger.info('Check rotated syslog file number')
+    num = duthost.shell('sudo ls -l /var/log | grep -E "syslog\\.[0-9]{1,4}[\\.gz]{0,1}" | wc -l')['stdout']
+    logger.debug('There are {} rotated syslog files'.format(num))
+    return int(num)
+
+
+def create_temp_syslog_file(duthost, size):
+    """
+    Create a temp syslog file with specific size and
+    :param duthost: DUT host object
+    :param size: file size in MB or KB
```

Comment:
Please be specific wither it should be in KB or MB.

Reply:
ACK